scipy/special/tests/test_mpmath.py

```diff
@@ -579,24 +579,42 @@ def test_cospi_zeros():
 # ------------------------------------------------------------------------------
 
 @check_version(mpmath, '0.19')
-def test_dn_quarter_period():
-    def dn(u, m):
-        return sc.ellipj(u, m)[2]
-
-    def mpmath_dn(u, m):
-        return float(mpmath.ellipfun("dn", u=u, m=m))
-
+def test_ellipj():
+    def wrap_ellipj(u, m):
+        sn,cn,dn,phi = sc.ellipj(u,m)
+        return sn,cn,dn
+
+    def mpmath_ellipj(u, m):
+        sn = float(mpmath.ellipfun("sn",u=u,m=m))
+        cn = float(mpmath.ellipfun("cn",u=u,m=m))
+        dn = float(mpmath.ellipfun("dn",u=u,m=m))
+        return sn,cn,dn
+
+    # Add the points that will be calculated using
+    # the AGM method
     m = np.linspace(0, 1, 20)
     du = np.r_[-np.logspace(-1, -15, 10), 0, np.logspace(-15, -1, 10)]
     dataset = []
     for m0 in m:
         u0 = float(mpmath.ellipk(m0))
         for du0 in du:
             p = u0 + du0
-            dataset.append((p, m0, mpmath_dn(p, m0)))
-    dataset = np.asarray(dataset)
+            dataset.append((p, m0)+mpmath_ellipj(p, m0))
 
-    FuncData(dn, dataset, (0, 1), 2, rtol=1e-10).check()
+    # Add the points that will use the m near 1 method
+    m = 0.99999999999
```

Comment:
it's probably worth going a step further. In addition to the one right at the threshold (0.9999999999), how about one just outside it (0.9999999998) and beyond it (0.99999999999)? It helps validate the choice of threshold.

Reply:
Is it better to write `1. - 1e-10`  for readability of the digits?

Reply:
Sounds good to me.

Reply:
fixed